tesseract/tesseract_collision/test/collision_sphere_sphere_unit.cpp

```diff
@@ -244,67 +253,43 @@ void runConvexTest(DiscreteContactManager& checker)
   EXPECT_NEAR(result_vector[0].nearest_points[0][2], result_vector[0].nearest_points[1][2], 0.001);
   EXPECT_GT((idx[2] * result_vector[0].normal).dot(Eigen::Vector3d(1, 0, 0)), 0.0);
   EXPECT_LT(std::abs(std::acos((idx[2] * result_vector[0].normal).dot(Eigen::Vector3d(1, 0, 0)))), 0.00001);
+}
 
+void runConvexTest3(DiscreteContactManager& checker)
+{
   //////////////////////////////////////////////////////////////////////
-  // Test when object is in collision (Closest Feature Vertex to Vertex)
+  // Test when object is in collision (Closest Feature face to edge)
   //////////////////////////////////////////////////////////////////////
+  tesseract_common::TransformMap location;
   location["sphere1_link"] = Eigen::Isometry3d::Identity();
   location["sphere1_link"].translation()(1) = 0.2;
   checker.setCollisionObjectsTransform(location);
 
   // Perform collision check
-  result.clear();
-  result_vector.clear();
-
+  ContactResultMap result;
+  ContactResultVector result_vector;
   checker.contactTest(result, ContactTestType::CLOSEST);
   flattenResults(std::move(result), result_vector);
 
+  //Bullet: -0.280223 {0.0425563,0.2308753,-0.0263040} {-0.0425563, -0.0308753, 0.0263040}
+  //FCL:    -0.280223 {0.0425563,0.2308753,-0.0263040} {-0.0425563, -0.0308753, 0.0263040}
   EXPECT_TRUE(!result_vector.empty());
-  EXPECT_NEAR(result_vector[0].distance, -0.25, 0.035);
+  EXPECT_NEAR(result_vector[0].distance, -0.280223, 0.001);
 
-  idx = { 0, 1, 1 };
+  std::vector<int> idx = { 0, 1, 1 };
   if (result_vector[0].link_names[0] != "sphere_link")
     idx = { 1, 0, -1 };
 
-  EXPECT_NEAR(result_vector[0].nearest_points[idx[0]][1], 0.25, 0.03);
-  EXPECT_NEAR(result_vector[0].nearest_points[idx[1]][1], -0.05, 0.03);
-  EXPECT_NEAR(result_vector[0].nearest_points[0][0], result_vector[0].nearest_points[1][0], 0.10);
-  EXPECT_NEAR(result_vector[0].nearest_points[0][2], result_vector[0].nearest_points[1][2], 0.10);
-  EXPECT_GT((idx[2] * result_vector[0].normal).dot(Eigen::Vector3d(0, 1, 0)), 0.0);
+  EXPECT_NEAR(std::abs(result_vector[0].nearest_points[idx[0]][0]), 0.0425563, 0.001);
+  EXPECT_NEAR(result_vector[0].nearest_points[idx[0]][1], 0.2308753, 0.001);
+  EXPECT_NEAR(std::abs(result_vector[0].nearest_points[idx[0]][2]), 0.0263040, 0.001);
+  EXPECT_NEAR(std::abs(result_vector[0].nearest_points[idx[1]][0]), 0.0425563, 0.001);
+  EXPECT_NEAR(result_vector[0].nearest_points[idx[1]][1], -0.0308753, 0.001);
+  EXPECT_NEAR(std::abs(result_vector[0].nearest_points[idx[1]][2]), 0.0263040, 0.001);
+  EXPECT_NEAR(std::abs(idx[2] * result_vector[0].normal[0]), 0.3037316, 0.001);
+  EXPECT_NEAR(idx[2] * result_vector[0].normal[1], 0.9340783, 0.001);
+  EXPECT_NEAR(std::abs(idx[2] * result_vector[0].normal[2]), 0.1877358, 0.001);
```

Comment:
My only comment is that these seem like magic values. I assume these are correct.

Reply:
Yea I agree.

Reply:
Right now on some of the tests, if both bullet and fcl produces the same results I add the values into the test. In this case it is symmetric which is why I added the std::abs.